pkg/manifests/manifests.go

```diff
@@ -1451,7 +1451,7 @@ func (f *Factory) PrometheusUserWorkload(grpcTLS *v1.Secret) (*monv1.Prometheus,
 		p.Spec.LogLevel = f.config.ClusterMonitoringConfiguration.PrometheusUserWorkloadConfig.LogLevel
 	}
 
-	if f.config.ClusterMonitoringConfiguration.PrometheusUserWorkloadConfig.Retention != "" {
+	if f.config.ClusterMonitoringConfiguration.PrometheusUserWorkloadConfig.Retention != "" && f.config.ClusterMonitoringConfiguration.PrometheusUserWorkloadConfig.Retention != "15d" {
```

Comment:
I don't get it. We are setting this to "15d" by default in https://github.com/openshift/cluster-monitoring-operator/blob/master/pkg/manifests/config.go#L222, why checking if it is not "15d" here changes anything?

Reply:
Yes sorry for the lack of context here did it before going on PTO. So the problem is it will always override the user workload configuration coming from user ConfigMap a few lines above if we don't exclude the defaults applied for ClusterMonitoringConfiguration.PrometheusUserWorkloadConfig coming from cluster monitoring user workload configmap. We only want to override this if it's overriden by the user, as this still takes precedence before the user workload.

Reply:
Now it makes sense, thanks for explanation :+1: We definitely need a test for this as it is not obvious and we may hit a regression if we decide to change default retention value.

Reply:
I understood too now. Small implementation detail: lets use a `const` so it becomes obvious where the value comes from.

Reply:
Changed to a constant, so this will not hit a regression if we fix the default value. @paulfantom PTAL thanks!